tests/behat/graph_helper.php

```diff
@@ -0,0 +1,242 @@
+            ;
+        }
+    }
+
+    /**
+     * @param string $username
+     *
+     * @return string
+     */
+    private function get_user_id(string $username): string {
+        $client = $this->get_admin_client();
+        $response = $client->request(
+            'GET',
+            '/graph/v1.0/users'
+        );
+        $responsebody = json_decode($response["body"], true);
+        foreach ($responsebody["value"] as $createduser) {
+            if ($createduser["displayName"] === $username) {
+                return $createduser["id"];
+            }
+        }
+    }
+
+    /**
+     * @param string $user
+     *
+     * @return array
+     */
+    public function create_new_user(string $user) {
+        $username = $this->get_actual_username($user);
+        $email = \str_replace(["@", " "], "", $username) . '@owncloud.com';
+        $payload['onPremisesSamAccountName'] = $username;
+        $payload['passwordProfile'] = ['password' => $this->get_password_for_user($username)];
+        $payload['displayName'] = $username;
+        $payload['mail'] = $email;
+        $payload['accountEnabled'] = true;
+
+        $client = $this->get_admin_client();
+        return $client->request(
+            'POST',
+            '/graph/v1.0/users',
+            json_encode($payload)
+        );
+    }
+
+    /**
+     * @param string $user
+     * @param string $spacename
+     *
+     * @return array
+     */
+    public function get_space_by_name(string $user, string $spacename): array {
+        $space = [];
+        $spacename = ($spacename === "Personal") ? "personal" : "project";
+        $client = $this->get_client($user);
+        $response = $client->request('GET', 'graph/v1.0/drives');
+        $responsebody = json_decode($response['body']);
+        foreach ($responsebody->value as $value) {
+            if ($value->driveType === $spacename) {
+                $space["name"] = $value->name;
+                $space['id'] = $value->id;
+            }
+        };
+        return $space;
+    }
+
+    /**
+     * @param string $user
+     * @param string $resource
+     * @param string $space
+     *
+     * @return string
+     */
+    public function get_resource_id(string $user, string $resource, string $space): string {
+        $client = $this->get_client($user);
+        $spaceid = $this->get_space_by_name($user, $space)['id'];
+        $response = $client->request(
+            'PROPFIND',
+            "/remote.php/dav/spaces/$spaceid/$resource"
+        );
+        $resbody = simplexml_load_string($response['body']);
+        return $resbody->xpath('//oc:fileid')[0];
+    }
+
+    /**
+     * @param string $user
+     * @param TableNode $table
+     *
+     * @return array
+     */
+    public function send_share_invitation(string $user, TableNode $table): array {
+        $body = [];
+        $rows = $table->getRowsHash();
+        if ($rows['permissionsRole'] === 'Viewer') {
```

Comment:
what should happen with other roles? Or are we expecting the ID there for other roles?